### PR TITLE
Week4

### PR DESCRIPTION
**Pull request by yograi343**
Hi @anilrao316,

Week 4 assignments completed with below implementation:
1. Promises were implemented with handling all states (resolve, reject and fulfilled) without the use of JS promises.
2. Inheritance of class is implemented with use of class, super and extends. Also used a html file to print the attributes like name, age and subject from the object him where name and age is inherited from parent constructor Person.
3. Fibonacci series is implemented by use of iterators

https://user-images.githubusercontent.com/63515909/172047658-93cd0a5d-42ee-4787-8cf1-43905dba3a78.mp4

.

### CODE REVIEW

File: Week-4/assignment4.2/teacher.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+var Person = function() {};
+Person.prototype.initialize=function(name,age){
+    this.name=name;this.age=age;
+    };
+// TODO: create the class Teacher and a method teach
+//Document.setPrototypeOf(Teacher,Person);
+//"use strict";
+class Teacher extends Person{
```
Comment (anilrao316) [inline Week-4/assignment4.2/teacher.js]: we should be using prototype inheritence and not class based inheritence

File: Week-4/assigmnent4.1/random.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  
+      executionFunction(this.onResolve, this.onReject);
+    }
+  
+    then(handleSuccess) {
+      this.promiseChain.push(handleSuccess);
+  
+      return this;
+    }
+
```
Comment (anilrao316) [inline Week-4/assigmnent4.1/random.js]: implement finally

File: Week-4/assignment4.2/teacher.js (hunk 1 above)
Comment (yograi343) [reply Week-4/assignment4.2/teacher.js]: inheritance achieved using prototype inheritance.

File: Week-4/assigmnent4.1/random.js (hunk 2 above)
Comment (yograi343) [reply Week-4/assigmnent4.1/random.js]: finally implemented in the promise as well.

**Comment by anilrao316** [comment]
looks good!

**Pull request merged**